src/Sunburst.ts

```diff
@@ -768,13 +772,29 @@ export class Sunburst implements IVisual {
         }
     }
 
-    private wrapPathText(padding?: number): (slice: HierarchyRectangularNode<SunburstDataPoint>, index: number) => void {
-        const self = this;
-        return function () {
-            const selection: Selection<BaseType, any, BaseType, any> = d3Select(this);
-            const width = (<SVGPathElement>d3Select(selection.attr("xlink:href")).node()).getTotalLength();
-            self.wrapText(selection, padding, width);
-        };
+    private wrapPathText(text: string, i: number, properties: TextProperties, ellipsisWidth: number) {
+
+        let width = (<SVGPathElement>d3Select("#sliceLabel_" + i).node()).getTotalLength();
```

Comment:
🔕Maybe it will be better?

let width = (<SVGPathElement>d3Select(**`#sliceLabel_${i}'**).node()).getTotalLength() **|| 0**;